mesonbuild/interpreter/compiler.py

```diff
@@ -223,20 +225,20 @@ def version_method(self, args: T.List['TYPE_var'], kwargs: 'TYPE_kwargs') -> str
     def cmd_array_method(self, args: T.List['TYPE_var'], kwargs: 'TYPE_kwargs') -> T.List[str]:
         return self.compiler.exelist
 
-    def _determine_args(self, nobuiltins: bool,
-                        incdirs: T.List[build.IncludeDirs],
-                        extra_args: T.List[str],
+    def _determine_args(self, kwargs: 'BaseCompileKW',
                         mode: CompileCheckMode = CompileCheckMode.LINK) -> T.List[str]:
         args: T.List[str] = []
-        for i in incdirs:
+        for i in kwargs['include_directories']:
             for idir in i.to_string_list(self.environment.get_source_dir()):
                 args.extend(self.compiler.get_include_args(idir, False))
-        if not nobuiltins:
+        if not kwargs['no_builtin_args']:
             opts = self.environment.coredata.options
             args += self.compiler.get_option_compile_args(opts)
             if mode is CompileCheckMode.LINK:
                 args.extend(self.compiler.get_option_link_args(opts))
-        args.extend(extra_args)
+        if kwargs.get('werror', False):
+            args.extend(self.compiler.get_werror_args())
```

Comment:
Is there some reason to use `kwargs.get('werror', False)` here instead of `kwargs['werror']` like we do for all the other kwargs?

typed_kwargs guarantees this is always set to something, in this case defaulting to False anyway.

Reply:
That function is called by other methods that don't have that kwarg, like has_header().

Reply:
Ah right, that makes sense -- I should have looked closer.